snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2EffectiveTimeSlice.java

```diff
@@ -262,6 +260,21 @@ public void doImport(
 				}
 				
 				tx.commit(commitMessage);
```

Comment:
This method also call `register` on the CIS for the SCTIDs in the transaction, but only on "new" components (from the repository's point of view): https://github.com/b2ihealthcare/snow-owl/blob/49fa236926e0b0737a8495f6d9becb5f901c092d/snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2TransactionContext.java#L90-L99

To avoid duplication, it would be good to merge the two mechanisms into one in `Rf2TransactionContext` instead (register SCTID in CIS if component has no effective time, publish SCTID in CIS if effective time is set, new appearances in transaction only).

Reply:
Alina's feedback: looking at new components only in `Rf2TransactionContext` is also flawed. Existing components should also be inspected in case they have `effectiveTime` set on them, in which case they should also be published.